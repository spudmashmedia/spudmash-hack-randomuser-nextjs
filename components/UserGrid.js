/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Spudmash Media Pty Ltd
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

import React, { useState } from "react";
import { Avatar, Paper, Grid, Button, Typography } from "@material-ui/core";
import { v1 as uuidv1 } from "uuid";
import Loading from "./Loading";
import * as constants from "../common/constants";

let UserGrid = props => {
  const userData = props.data;
  const isLoading = props.isLoading;

  if (isLoading == true)
    return (
      <Grid
        container
        justify="center"
        spacing={1}
        direction="column"
        alignItems="center"
      >
        <Grid item>
          <Loading text={"Something is loading"} />
        </Grid>
        <Grid item>
          <Typography>Loading...</Typography>
        </Grid>
      </Grid>
    );

  return (
    <Paper elevation={3}>
      <Grid container justify="center" spacing={1}>
        {userData.map(userItem => (
          <Grid key={uuidv1()} item>
            <Button>
              <Avatar alt={userItem.name} src={userItem.thumbnail} />
            </Button>
          </Grid>
        ))}
      </Grid>
    </Paper>
  );
};

export default UserGrid;
