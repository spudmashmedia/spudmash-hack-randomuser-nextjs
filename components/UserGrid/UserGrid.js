/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Spudmash Media Pty Ltd
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

import React from "react";
import { Paper, Grid, Typography } from "@material-ui/core";
import { v1 as uuidv1 } from "uuid";
import LoadingWidget from "../LoadingWidget";
import Profile from "../Profile";

let UserGrid = props => {
  const userData = props.data;
  const isLoading = props.isLoading;

  if (isLoading == true)
    return (
      <Grid
        container
        justify="center"
        spacing={1}
        direction="column"
        alignItems="center"
      >
        <Grid item>
          <LoadingWidget text={"Something is loading"} />
        </Grid>
        <Grid item>
          <Typography>Loading...</Typography>
        </Grid>
      </Grid>
    );

  return (
    <Paper elevation={3}>
      <Grid container justify="center" spacing={1}>
        {userData.map(userItem => (
          <Grid key={`${userItem.id}-${uuidv1()}`} item>
            <Profile data={userItem} />
          </Grid>
        ))}
      </Grid>
    </Paper>
  );
};

export default UserGrid;
