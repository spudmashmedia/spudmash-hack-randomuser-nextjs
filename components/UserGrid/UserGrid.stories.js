/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Spudmash Media Pty Ltd
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

import React from "react";
import UserGrid from "./UserGrid";

export default {
  component: UserGrid,
  title: "UserGrid",
  excludeStories: /.*Data$/
};

export const actionData = [
  {
    id: "15667689-X",
    name: "Angela Moya",
    email: "angela.moya@example.com",
    phone: "929-044-312",
    location: "Orihuela, Canarias Spain",
    image_thumb: "https://randomuser.me/api/portraits/thumb/women/73.jpg",
    image_mid: "https://randomuser.me/api/portraits/med/women/73.jpg",
    image_lrg: "https://randomuser.me/api/portraits/women/73.jpg"
  },
  {
    id: "2NNaN36732140 24",
    name: "Ambre Deschamps",
    email: "ambre.deschamps@example.com",
    phone: "03-81-45-45-60",
    location: "Aubervilliers, Guyane France",
    image_thumb: "https://randomuser.me/api/portraits/thumb/women/72.jpg",
    image_mid: "https://randomuser.me/api/portraits/med/women/72.jpg",
    image_lrg: "https://randomuser.me/api/portraits/women/72.jpg"
  },
  {
    id: "991228915",
    name: "Andrew Young",
    email: "andrew.young@example.com",
    phone: "03-1017-9496",
    location: "Australian Capital Territory, New South Wales Australia",
    image_thumb: "https://randomuser.me/api/portraits/thumb/men/92.jpg",
    image_mid: "https://randomuser.me/api/portraits/med/men/92.jpg",
    image_lrg: "https://randomuser.me/api/portraits/men/92.jpg"
  },
  {
    id: "64857192-J",
    name: "Miguel Nieto",
    email: "miguel.nieto@example.com",
    phone: "950-858-588",
    location: "Lorca, RegiÃ³n de Murcia Spain",
    image_thumb: "https://randomuser.me/api/portraits/thumb/men/20.jpg",
    image_mid: "https://randomuser.me/api/portraits/med/men/20.jpg",
    image_lrg: "https://randomuser.me/api/portraits/men/20.jpg"
  },
  {
    id: "77116352",
    name: "Marc Van Breemen",
    email: "marc.vanbreemen@example.com",
    phone: "(724)-143-6221",
    location: "Rijsoord, Drenthe Netherlands",
    image_thumb: "https://randomuser.me/api/portraits/thumb/men/65.jpg",
    image_mid: "https://randomuser.me/api/portraits/med/men/65.jpg",
    image_lrg: "https://randomuser.me/api/portraits/men/65.jpg"
  },
  {
    id: "00000001",
    name: "Not A Real Guy",
    email: "not.a.real@guy.com",
    phone: "+61414000111",
    location: "Sydney, NSW Australia",
    image_thumb: "",
    image_mid: "",
    image_lrg: ""
  }
];

export const isLoadingTrue = () => <UserGrid isLoading={true} />;

export const dataLoaded = () => (
  <UserGrid data={actionData} isLoading={false} />
);
