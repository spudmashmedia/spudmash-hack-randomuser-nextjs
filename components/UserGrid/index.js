/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Spudmash Media Pty Ltd
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

import UserGrid from "./UserGrid";
export default UserGrid;
