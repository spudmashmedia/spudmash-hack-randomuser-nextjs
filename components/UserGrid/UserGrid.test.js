import React from "react";
import UserGrid from "./UserGrid";
import { actionData } from "./UserGrid.stories";
import renderer from "react-test-renderer";

describe("UserGrid Component Snapshot Test", () => {
  test("renders correctly - loading", () => {
    const tree = renderer.create(<UserGrid isLoading={true} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
  test("renders correctly - with data", () => {
    const tree = renderer
      .create(<UserGrid data={actionData} isLoading={false} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
