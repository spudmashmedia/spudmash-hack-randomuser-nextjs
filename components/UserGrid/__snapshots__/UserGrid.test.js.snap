// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserGrid Component Snapshot Test renders correctly - loading 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      Loading...
    </p>
  </div>
</div>
`;

exports[`UserGrid Component Snapshot Test renders correctly - with data 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation3 MuiPaper-rounded"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div>
        <button
          aria-controls={null}
          aria-haspopup={true}
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          id="btn-15667689-X"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="15667689-X:[Angela Moya]"
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <div
              className="MuiAvatar-root MuiAvatar-circle"
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
              style={
                Object {
                  "transform": "perspective(600px) rotateX(0deg) rotateY(0deg) scale(1)",
                }
              }
            >
              <img
                alt="Angela Moya"
                className="MuiAvatar-img"
                src="https://randomuser.me/api/portraits/thumb/women/73.jpg"
              />
            </div>
          </span>
        </button>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div>
        <button
          aria-controls={null}
          aria-haspopup={true}
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          id="btn-2NNaN36732140 24"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="2NNaN36732140 24:[Ambre Deschamps]"
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <div
              className="MuiAvatar-root MuiAvatar-circle"
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
              style={
                Object {
                  "transform": "perspective(600px) rotateX(0deg) rotateY(0deg) scale(1)",
                }
              }
            >
              <img
                alt="Ambre Deschamps"
                className="MuiAvatar-img"
                src="https://randomuser.me/api/portraits/thumb/women/72.jpg"
              />
            </div>
          </span>
        </button>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div>
        <button
          aria-controls={null}
          aria-haspopup={true}
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          id="btn-991228915"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="991228915:[Andrew Young]"
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <div
              className="MuiAvatar-root MuiAvatar-circle"
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
              style={
                Object {
                  "transform": "perspective(600px) rotateX(0deg) rotateY(0deg) scale(1)",
                }
              }
            >
              <img
                alt="Andrew Young"
                className="MuiAvatar-img"
                src="https://randomuser.me/api/portraits/thumb/men/92.jpg"
              />
            </div>
          </span>
        </button>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div>
        <button
          aria-controls={null}
          aria-haspopup={true}
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          id="btn-64857192-J"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="64857192-J:[Miguel Nieto]"
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <div
              className="MuiAvatar-root MuiAvatar-circle"
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
              style={
                Object {
                  "transform": "perspective(600px) rotateX(0deg) rotateY(0deg) scale(1)",
                }
              }
            >
              <img
                alt="Miguel Nieto"
                className="MuiAvatar-img"
                src="https://randomuser.me/api/portraits/thumb/men/20.jpg"
              />
            </div>
          </span>
        </button>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div>
        <button
          aria-controls={null}
          aria-haspopup={true}
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          id="btn-77116352"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="77116352:[Marc Van Breemen]"
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <div
              className="MuiAvatar-root MuiAvatar-circle"
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
              style={
                Object {
                  "transform": "perspective(600px) rotateX(0deg) rotateY(0deg) scale(1)",
                }
              }
            >
              <img
                alt="Marc Van Breemen"
                className="MuiAvatar-img"
                src="https://randomuser.me/api/portraits/thumb/men/65.jpg"
              />
            </div>
          </span>
        </button>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div>
        <button
          aria-controls={null}
          aria-haspopup={true}
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          id="btn-00000001"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="00000001:[Not A Real Guy]"
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <div
              className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
              style={
                Object {
                  "transform": "perspective(600px) rotateX(0deg) rotateY(0deg) scale(1)",
                }
              }
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root MuiAvatar-fallback"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
